sql/country_statistics.sql

```diff
@@ -1,117 +1,140 @@
-          sum(chg.pois_added) AS pois_added,
-          sum(chg.pois_modified + chg.pois_deleted) AS pois_modified,
-          max(coalesce(chg.closed_at, chg.created_at)) AS last_edit,
-          max(COALESCE(chg.closed_at, chg.created_at, chg.updated_at)) AS updated_at,
-          count(*) AS changeset_count,
-          sum(cc.edit_count) AS edit_count
-        FROM (changesets chg
-          JOIN country_counts cc ON ((cc.changeset_id = chg.id)))
-        GROUP BY cc.id
-    )
-  SELECT agg.country_id,
-      countries.name AS country_name,
-      countries.code AS country_code,
-      agg.road_km_added,
-      agg.road_km_modified,
-      agg.waterway_km_added,
-      agg.waterway_km_modified,
-      agg.coastline_km_added,
-      agg.coastline_km_modified,
-      agg.roads_added,
-      agg.roads_modified,
-      agg.waterways_added,
-      agg.waterways_modified,
-      agg.coastlines_added,
-      agg.coastlines_modified,
-      agg.buildings_added,
-      agg.buildings_modified,
-      agg.pois_added,
-      agg.pois_modified,
-      agg.last_edit,
-      agg.updated_at,
-      agg.changeset_count,
-      agg.edit_count,
-      jce.edits AS user_edit_counts,
-      hts.hashtag_edits
-    FROM (agg_stats agg
-      FULL OUTER JOIN json_country_edits jce ON (agg.country_id = jce.country_id)
-      FULL OUTER JOIN ht_json hts ON (agg.country_id = hts.country_id)
-      JOIN countries ON agg.country_id = countries.id);
+  WITH changesets AS (
+    SELECT
+      *
+    FROM changesets
+    -- ignore users 0 and 1
+    WHERE user_id > 1
+  ),
+  general AS (
+    SELECT
+      country_id,
+      max(coalesce(closed_at, created_at)) last_edit,
+      count(*) changeset_count,
+      sum(edit_count) edit_count,
+      max(updated_at) updated_at
+    FROM changesets
+    JOIN changesets_countries ON changesets.id = changesets_countries.changeset_id
+    GROUP BY country_id
+  ),
+  processed_changesets AS (
+    SELECT
+      -- TODO include changesets_countries.edit_count as an alternative to changeset count
+      id,
+      user_id,
+      country_id,
+      measurements,
+      counts
+    FROM changesets
+    JOIN changesets_countries ON changesets.id = changesets_countries.changeset_id
+  ),
+  hashtag_counts AS (
+    SELECT
+      -- TODO rank by edit count?
+      RANK() OVER (PARTITION BY country_id ORDER BY count(*) DESC) AS rank,
+      country_id,
+      hashtag,
+      -- TODO expose edit count instead?
+      count(*) changesets
+    FROM processed_changesets
+    JOIN changesets_hashtags ON processed_changesets.id = changesets_hashtags.changeset_id
+    JOIN hashtags ON changesets_hashtags.hashtag_id = hashtags.id
+    GROUP BY country_id, hashtag
+  ),
+  hashtags AS (
+    SELECT
+      country_id,
+      jsonb_object_agg(hashtag, changesets) hashtags
+    FROM hashtag_counts
+    WHERE rank <= 10
+    GROUP BY country_id
+  ),
+  user_counts AS (
+    SELECT
+      -- TODO rank by edit count?
```

Comment:
@giblet22 this produces the top 10 users by number of changesets (and surfaces that count in the response). Is that the correct thing to sort by? (It could also make sense to sort by number of edits.)

Reply:
Is this what is being surfaced for users in the country pages on Scoreboard? I believe scoreboard has them sorted by number of edits. cc: @kamicut

Reply:
I'm not sure. @kamicut do you know where Scoreboard gets / uses that list?